### ISSUE TITLE
Sticky Popups

### ISSUE THREAD

**Issue by brotatos** [python-visualization/folium#784]
On click, a popup will appear over a marker. However, this popup disappears when you click on another marker.

Please add the option to make a popup sticky (i.e. clicking on other markers does not affect whether a sticky popup remains over a map).

e.g.
```python
folium.Marker(
    popup=folium.Popup(..., sticky=True),
    ....
)
```

**Comment by Conengmo** [comment]
I just tried this out, it can be done using the `autoClose` and `closeOnClick` options of Leaflet's `Popup` class. Interestingly, @jwhendy just implemented these in PR #778. So maybe we can combine these two functionalities? 

Something like `sticky` adding the `autoClose` and `closeOnClick` options, and `show` also adding those while also opening the popup on load.

**Comment by jwhendy** [comment]
@Conengmo and @brotatos seems like some users simultaneously landed on these sorts of features being something desirable!

In the PR, I was trying to think through what sort of options we would want to open to the user. At present, my use-case was simply having them open on load. Since I'm using both of the `leaflet` options you mention to produce this effect, I think splitting them out would be pretty easy.

That PR is still waiting on me to write some tests; I could amend it to break these out... if I understand your suggestion, I think it sounds great would would do this:like:

- `sticky`: add `autoClose` and `closeOnClick` options
- `show`: both of the above + calling `.openPopup()`

**Comment by Conengmo** [comment]
Thanks for picking this up @jwhendy! I think having sticky popups can be useful, so nice that you're considering splitting them.

One remark, I'm not sure about `show` also calling `autoClose` and `closeOnClick`. I'm assuming the defaults for `sticky` and `show` will be `False`. 

- If someone explicitly sets `show=True` and `sticky=False`, that should not be overruled I think. For some people there may be a use-case to show the popups on load for a quick inspection, but also being able to close them all together after that. 

- If someone only sets `show=True`, it's maybe confusing that the popups become sticky, while the default for that is `sticky=False`.

- There is no way to distinguish between `sticky=False` being set by the default, or being set explicitly by the user.

- We'd have to communicate to the user that setting `show=True` makes `sticky` unusable.

So I would make the case for:

- `sticky`: add `autoClose` and `closeOnClick` options
- `show`: call `.openPopup()`

What are your thoughts on this?

**Comment by jwhendy** [comment]
Good points, and honestly I'm not sure. Take a look at the first post in #778 and my end checklist. I wasn't sure myself what people would want when I made that PR and asked:

>should other variables be added in order to allow users more granularity, giving access to `autoClose`, `closeOnClick`, and `closePopupOnClick`?

Maybe you already know how these options behave, but I just made some examples for others wanting to see. I forked these from a fiddle I found at a relevant [leaflet issue](https://github.com/Leaflet/Leaflet/issues/5719).

- `autoClose` ([link](https://jsfiddle.net/6yfuxdjp/3/)): by itself, not useful. From the linked issue on `leaflet`, you'll note that clicking a popup is still clicking on the map, so clicking one closes others.
- `autoClose, closeOnClick` ([link](https://jsfiddle.net/upwv5ayw/1/)): this is what I think we agree that `sticky` behavior should be. Popup toggled only by clicks, not what's done to other popups/map.
- `openPopup()` ([link](https://jsfiddle.net/a5p3wrha/1/)): this won't work by itself; each subsequent `openPopup()` call appears to behave like a click and close the previous ones. The only open popup on page load will be the nth you added
- `openPopup()`, `autoClose` ([link](https://jsfiddle.net/de8d1qux/1/)): I *think* this is what you're suggesting the `show` behavior would be. Still not *totally* clear on how leaflet options are intended, so clicking one the first time won't close the other, but opening it back up *will* close others. Clicking the map will also close all that are open.
- `openPopup()`, `closeOnClick`: suffers from the same issue of just calling `openPopup()`; only the last rendered popup will show. Also, clicking one popup will still close others, so this seems like a non-usefu combo
- `openPoup()`, `autoClose`, `closeOnClick` ([link](https://jsfiddle.net/3hta0sj7/1/)): how I currently wrote my `show` option, but is basically `sticky` and `show` how you're describing.

When you first brought up splitting these out, my reaction was initially that for my use case, I was thinking I wanted both of these sorts of behaviors (open on load, don't close on click). Hence, those would be built into `show`. I have no idea why it didn't dawn on me to just call *both* in my case!

My new vote would be:

- `show`: `autoClose` and `openPopup()`. Popups will start open on page load, but toggling one of them, or simply clicking the map will make them go away. That'd be [this behavior](https://jsfiddle.net/de8d1qux/1/)
- `sticky`: `autoClose` and `closeOnClick`. Think this is clear enough. That would be [this behavior](https://jsfiddle.net/upwv5ayw/1/)

I just implemented this on [my `open-popup` branch](https://github.com/jwhendy/folium/tree/open-popup), but haven't updated my PR yet. Want to try it and see if that behaves? I just tried it with no options, `sticky` and `show` separately, and combined, and think it's working nicely! Thanks for the dialog/suggestions; I like this route.

**Comment by Conengmo** [comment]
I think you researched this really well. You're right, `show` should also have `autoClose=False`. Your current proposal is good I think!

**Comment by Conengmo** [comment]
Merged in #778
